modules/asset-manager-storage-aws/src/main/java/org/opencastproject/assetmanager/aws/s3/AwsS3AssetStore.java

```diff
@@ -283,11 +335,103 @@ protected AwsUploadOperationResult uploadObject(File origin, String objectName)
     }
   }
 
+  /**
+   * Return the object key of the asset in S3
+   * @param storagePath asset storage path
+   */
+  public String getAssetObjectKey(StoragePath storagePath) throws AssetStoreException {
+    try {
+      AwsAssetMapping map = database.findMapping(storagePath);
+      return map.getObjectKey();
+    } catch (AwsAssetDatabaseException e) {
+      throw new AssetStoreException(e);
+    }
+  }
+
+  /**
+   * Return the storage class of the asset in S3
+   * @param storagePath asset storage path
+   */
+  public String getAssetStorageClass(StoragePath storagePath) throws AssetStoreException {
+    if (!contains(storagePath)) {
+      return "NONE";
+    }
+    return getObjectStorageClass(getAssetObjectKey(storagePath));
+  }
+
+  private String getObjectStorageClass(String objectName) throws AssetStoreException {
+    try {
+      String storageClass = s3.getObjectMetadata(bucketName, objectName).getStorageClass();
+      return storageClass == null ? StorageClass.Standard.toString() : storageClass;
+    } catch (SdkClientException e) {
+      throw new AssetStoreException(e);
+    }
+  }
+
+  /**
+   * Change the storage class of the object if possible
+   * @param storagePath asset storage path
+   * @param storageClassId metadata storage class id
+   * @see <a href="https://aws.amazon.com/s3/storage-classes/">The S3 storage class docs</a>
+   */
+  public String modifyAssetStorageClass(StoragePath storagePath, String storageClassId) throws AssetStoreException {
+    try {
+      StorageClass storageClass = StorageClass.fromValue(storageClassId);
+      AwsAssetMapping map = database.findMapping(storagePath);
+      return modifyObjectStorageClass(map.getObjectKey(), storageClass).toString();
+    } catch (AwsAssetDatabaseException | IllegalArgumentException e) {
+      throw new AssetStoreException(e);
+    }
+  }
+
+  private StorageClass modifyObjectStorageClass(String objectName, StorageClass storageClass)
+          throws AssetStoreException {
+    try {
+      StorageClass objectStorageClass = StorageClass.fromValue(getObjectStorageClass(objectName));
+
+      if (storageClass != objectStorageClass) {
+        /* objects can only be retrieved from Glacier not moved */
+        if (objectStorageClass == StorageClass.Glacier || objectStorageClass == StorageClass.DeepArchive) {
+          logger.warn("S3 Object {} can not be moved from storage class {}", objectStorageClass);
```

Comment:
message is missing the objectName argument

Reply:
Fixed!